digest/src/lib.rs

```diff
@@ -173,3 +171,67 @@ pub trait ExtendableOutput {
         buf
     }
 }
+
+/// Trait for variable output size hash functions.
+pub trait VariableOutput: Sized {
```

Comment:
I think it'd be nice if this trait had a method like `Digest::digest`, something like:

```rust
fn digest_variable(input: &[u8], output: &mut [u8]) -> Result<(), InvalidOutputSize> {
    let hasher = Self::new(output.len())?;
    hasher.update(input);
    hasher.finalize_variable(|out| output.copy_from-slice(out));
    Ok(())
}
```

...which could be used as a one-shot API

Reply:
Done!

Reply:
Sweet, thank you!